src/main/java/org/opensearch/security/configuration/ConfigurationRepository.java

```diff
@@ -101,7 +102,7 @@ public class ConfigurationRepository implements ClusterStateListener {
     private final Client client;
     private final Cache<CType<?>, SecurityDynamicConfiguration<?>> configCache;
     private final List<ConfigurationChangeListener> configurationChangedListener;
-    private final ConfigurationLoaderSecurity7 cl;
+    private ConfigurationLoaderSecurity7 cl;
```

Comment:
While I am generally quite open for `@VisibleForTesting` methods, I am struggling a bit making a `final` object attribute non-final in order to enable testing. `final` object attributes provide great value, as the assurance helps with thread safety and the general robustness of the code.

I would suggest to change the constructor of `ConfigurationRepository` instead and give the possibilty to pass a `ConfigurationLoader` instance there. That would allow you to retain the `final` modifier.

Reply:
Sure, will update.